app/models/ingredient.rb

```diff
@@ -1,3 +1,5 @@
 class Ingredient < ApplicationRecord
    validates_presence_of :name, :cost
+   has_many :recipe_ingredients
+   has_many :recipes, through: :recipe_ingredients, source: :recipe
```

Comment:
Using `source` here is not necessary but I'm curious how you were hoping to use the source keyword?